### ISSUE TITLE
Poor accuracy with Pytorch reimplementation

### ISSUE THREAD

**Issue by davidtranno1** [mahmoudnafifi/11K-Hands#3]
Hi @mahmoudnafifi ,
Many thanks for your great job. 

I wanted to run your code to verify the proposed results. However I faced a huge problem when could not run Matlab because of licence issues. I also tried Octave as an alternative but not success.

Finally, I decided to reproduce your main idea via open source Pytorch. Unfortunately, the results is not good as I expected. The first try is gender classification with CNN/AlexNet, as mentioned in your paper the highest accuracy could be 0.839. I tried to fine tune the pretrained AlexNet in the some scenarios but fails:

- Freeze Convolution layers and add two new FCs (with 2048 and 531 units respectively with xavier init), I only train to tweak FC layers (included FC6 and FC7 of Alexnet)

- Retrain Convolution layers and FCs layers (after loading pretrained weights)

(I also implemented your dual channel but the result not good, so I tried to reproduce with normal CNN as a base line with hoping that the accuracy should greater than 0.8)

However the highest accuracy I can achieve is 0.6 (I conducted experiment with palm size first).
Please correct me if I wrong.
Thank you!

**Comment by mahmoudnafifi** [comment]
I did not try it on Pytorch. The training parameters are described in the source code, also in the paper: https://arxiv.org/pdf/1711.04322.pdf (sec 6.1).

For AlexNet or similar pre-trained models, I used the pre-trained weights as it and replaced the last fully connected layer with a new one with two neurons (male/female). In training, the learning rate for the entire architecture was λ = 10−4, except for the new FC layer it was 0.002.

I hope it helps!
